src/clinical/clinical-service.ts

```diff
@@ -289,13 +292,65 @@ export const getClinicalSearchResults = async (
 };
 
 export const getClinicalErrors = async (programId: string, donorIds: number[]) => {
-  // 1. Get the migration errors for every donor requested...
+  // 1. Get the migration errors for every donor requested
   const migrationErrors = await getClinicalEntityMigrationErrors(programId, donorIds);
 
-  // 2. ...and now remove from the list all valid donors (fixed with submissions since the migration)
-  const validErrorRecords = await getValidRecordsPostSubmission(programId, migrationErrors);
+  // 2. Remove from the list all valid donors (fixed with submissions since the migration)
+  const validPostSubmissionErrors = await getValidRecordsPostSubmission(programId, migrationErrors);
+
+  // 3. Remove from the list all errors which match Program Exceptions
+  const programExceptions = await exceptionService.operations.getProgramException({ programId });
+  const entityExceptions = await exceptionService.operations.getEntityException({ programId });
+  const hasProgramExceptions = 'exception' in programExceptions;
+  const hasEntityExceptions = 'exception' in entityExceptions;
+
+  if (
+    featureFlags.FEATURE_SUBMISSION_EXCEPTIONS_ENABLED &&
+    (hasProgramExceptions || hasEntityExceptions)
+  ) {
+    const programRecords = hasProgramExceptions ? programExceptions.exception.exceptions : [];
+    const entityRecords = hasEntityExceptions
+      ? [...entityExceptions.exception.specimen, ...entityExceptions.exception.follow_up]
+      : [];
+
+    const exceptionRecords = [...programRecords, ...entityRecords];
+
+    const clinicalErrors = validPostSubmissionErrors.clinicalErrors.map(errorRecord => {
+      const matchedExceptions = exceptionRecords.filter(
+        exception =>
+          exception.schema === errorRecord.entityName &&
+          errorRecord.errors.some(error => error.fieldName === exception.requested_core_field),
+      );
+      // If no matching exceptions related to error, error is included in the result
+      const hasExceptions = matchedExceptions.length;
+      if (!hasExceptions) return errorRecord;
+
+      // Remove Error records where both field and value match the exception
+      const errors = errorRecord.errors.filter(error => {
+        const hasErrorValue = 'value' in error.info;
+
+        // Lectern client is not returning specific error values for every entry
```

Comment:
... but it should be, needs follow up ticket

Reply:
See https://github.com/overture-stack/js-lectern-client/blob/master/src/schema-functions.ts#L629

Reply:
Appreciate how far you've looked into this. Could you provide more explanation of what is wrong in the lectern client? Is this something we can fix in the client? More explanation of the problem would help understand the next action.

Reply:
We found yesterday, Lectern Client wasn't truly causing an issue; but, there will be some cases where we may be unable to be 100% accurate labelling errors on the UI. When error info/value are not provided, we can leverage the `index` property, but will have to see how reliably that works.